index.src.html

```diff
@@ -1720,7 +1720,9 @@
   constructor(EncodedAudioChunkInit init);
   readonly attribute EncodedAudioChunkType type;
   readonly attribute unsigned long long timestamp;  // microseconds
-  readonly attribute ArrayBuffer data;
+  readonly attribute unsigned long byteLength;
+
+  undefined readInto(ArrayBufferView dst);
```

Comment:
`copyTo` is easier to understand because it says what it does. The naming is also in line with what the Web Audio API has, and someone close to `TypedArray.copyWithin`.

Reply:
@sandersdan - lets chat

Reply:
`readInto()` is what I've proposed for `VideoFrame` readback. There the name expresses two things: it's not necessarily just a copy, and the destination is an app-provided buffer rather than a new one.

This use in chunks only has the second property.

In use:

```js
let layout = await frame.readInto(buf);
chunk.readInto(buf);

let layout = await frame.copyTo(buf);
chunk.copyTo(buf);
```

In context prefer `readInto()` for frame readback, but `copyTo()` for chunks (and I don't like `readTo()` at all). I'm not sure what the most consistent set of choices is.

Reply:
> it's not necessarily just a copy

IIUC, the point you're making here is in cases like texture backed frames where read is doing things like unswizzling etc... 

I follow that this is not just a copy. Is this an important signal to the JS user? If so, I guess we split the baby, keeping readInto() for VideoFrame, but copyTo() for chunks. Otherwise, we can make them both copyTo()... but read on. 

> WebAudio has copyToBuffer()

This was wrong. WebAudio [actually has](https://webaudio.github.io/web-audio-api/#audiobuffer) 

```
interface AudioBuffer {
  ...
  undefined copyFromChannel (Float32Array destination,
                             unsigned long channelNumber,
                             optional unsigned long bufferOffset = 0);
  undefined copyToChannel (Float32Array source,
                           unsigned long channelNumber,
                           optional unsigned long bufferOffset = 0);
};
```

So there, copyTo*() is actually a mutation API. It's copyFrom*() that is the "read". But we definitely don't want copyFrom(buffer) here... as that is even more hinting at the wrong direction. WebAudio only gets by because its copyFromChannel(). And I just copied that name in PR #162.

Best I've come up with so far: copyOut(buffer). "Out" is pretty unambiguous about the direction. We can have VideoFrame name it readOut(buffer) if that distinction matters. 

Aside: in other discussion, we've floated the idea of adding a copy() API to VideoFrame that makes a deep copy. We'd have to name this something else if we change readInto() -> to be some sort of copy*().

Reply:
Internally, my idea of `copyOut()` didn't get much love, while `copyTo()` and `copyInto()` were both deemed acceptable. 

On this thread, `copyTo()` has been preferred to `copyInto()`. I'm supportive of that. Note that it is backward from WebAudio (copyToChannel() is a mutator) , but I think WebAudio might be in the wrong. It's good for the name to be intuitive as you write/read the code from left to right. That is `chunk.copyTo(buffer)` reads as "chunk: copy your contents to buffer". This is consistent with Streams (pipeTo()) and TextEncoder (encodeInto()). 

So I'm joining the `copyTo()` camp. Any objections?

Re: `readInto()`, we would make this simply `readTo()` or `copyTo()`. I lean `copyTo()` here as well. I take @sandersdan's point about `readInto()` suggesting that it's more than just a copy (i.e. it may be an expensive async readback, involving unswizzling etc)... but I don't think "read" really hints at that very strongly ([making frame.readInto() async](https://github.com/w3c/webcodecs/pull/208#discussion_r625508233) is probably the stronger hint). @sandersdan any objection?

Reply:
I interpret "read" as potentially expensive since the term for copying an image from GPU memory is "readback", and also by association with WebGL `readPixels()` (which presumably got its name for the previous reason).

No objections to any of these name, though.

Reply:
Understood. 

In summary
1. we're aligned on copyTo() for all non-VideoFrame things (including chunks)
2. we're ambivalent about copyTo() vs readTo() for VideoFrame. 

Re point 2, @padenot would you like to be the tie breaker?

Reply:
I like @sandersdan's argument of using `read` because of `readback` and `glRead{,n}Pixels` (not only in WebGL, also in regular GL).

I also still like 1. 

But this discussion is about `Encoded*Chunk` so this discussion doesn't need to block merging this. It's not the same operation at all.